Minio/Helper/Utils.cs

```diff
@@ -196,6 +197,20 @@ internal static bool IsSupersetOf(IList<string> l1, IList<string> l2)
         return !l2.Except(l1).Any();
     }
 
+    public static Task ForEachAsync<T>(this IEnumerable<T> source, int dop, Func<T, Task> body)
```

Comment:
Just take in the list of Tasks directly and depending on the .NET platform wrap it in a Task.WhenAll or Parallel.ForEachAsync